### ISSUE TITLE
TIMEOUT resulting in quality of 0.0

### ISSUE THREAD

**Issue by mfeurer** [sfalkner/pySMAC#7]
Hi,

I'm trying to tune a target algorithm for quality with a time limit for each function call. This results in SMAC return an error of 0.0 in case one function call hits the time limit. The following adaption of the example demonstrates the problem:

```
from __future__ import print_function, division

import pysmac

import sklearn.ensemble
import sklearn.datasets
import sklearn.cross_validation

X,Y = sklearn.datasets.make_classification(100000, 20, random_state=2)
X_train, X_test, Y_train, Y_test = sklearn.cross_validation.train_test_split(X,Y, test_size=0.33, random_state=1)

def random_forest(n_estimators,criterion, max_features, max_depth):

    predictor = sklearn.ensemble.RandomForestClassifier(n_estimators, criterion, max_features, max_depth)
    predictor.fit(X_train, Y_train)

    return -predictor.score(X_test, Y_test)

parameter_definition=dict(\
    max_depth   =("integer", [1,10],    4),
    max_features=("integer", [1,20],   10),
    n_estimators=("integer", [10,100], 10, 'log'),
    criterion   =("categorical", ['gini', 'entropy'], 'entropy'),
    )

opt = pysmac.SMAC_optimizer(working_directory = '/tmp/pysmac_test/',
                            persistent_files=False, debug = False)

predictor = sklearn.ensemble.RandomForestClassifier()
predictor.fit(X_train, Y_train)
print(('The default accuracy is %f'%predictor.score(X_test, Y_test)))


# The minimize method also has optional arguments
value, parameters = opt.minimize(random_forest,
                    100 , parameter_definition,
                    num_runs = 2,
                    seed = 2,
                    num_procs = 2,
                    mem_limit_function_mb=1000,
                    t_limit_function_s = 2)


print(('The highest accuracy found: %f'%(-value)))
print(('Parameter setting %s'%parameters))
```

I think for optimizing quality, a TIMEOUT not result in a good evaluation of a configuration but should have the same quality as a CRASH.

**Comment by sfalkner** [comment]
Hmmm...I haven't tested that case, but the quality of a timeout should not matter at all. Could it be that all configs hit the time limit? If so, SMAC will just return the default config, and the corresponding value 0 although this is a timeout. I will look into that.

**Comment by sfalkner** [comment]
hopefully fixed by commit d8ddaded97c94a566f1605fef55899c81cc7682f
now pysmac has an additional option called 'timeout_quality' that lets you define a value for all timeouts.
